RecoTracker/MkFitCMS/interface/LayerNumberConverter.h

```diff
@@ -0,0 +1,172 @@
+        if (cmsswdet == 5 && cmsswlay == 5 && isStereo == 0)
+          return 15 + lOffset;
+        if (cmsswdet == 5 && cmsswlay == 6 && isStereo == 0)
+          return 16 + lOffset;
+        return -1;
+      }
+    }
+    int convertDiskNumber(int cmsswdet, int cmsswdisk, bool useMatched, int isStereo) const {
+      if (cmsswdet == 1 || cmsswdet == 3 || cmsswdet == 5)
+        return -1;  //BPIX, TIB, TOB
+      if (cmsswdet == 2)
+        return cmsswdisk - 1;  //FPIX
+      int lOffset = 0;
+      if (lo_ == TkLayout::phase1)
+        lOffset = 1;
+      if (useMatched) {
+        return -1;
+      } else {
+        //TID
+        if (cmsswdet == 4 && cmsswdisk == 1 && isStereo == 0)
+          return 2 + lOffset;
+        if (cmsswdet == 4 && cmsswdisk == 1 && isStereo == 1)
+          return 3 + lOffset;
+        if (cmsswdet == 4 && cmsswdisk == 2 && isStereo == 0)
+          return 4 + lOffset;
+        if (cmsswdet == 4 && cmsswdisk == 2 && isStereo == 1)
+          return 5 + lOffset;
+        if (cmsswdet == 4 && cmsswdisk == 3 && isStereo == 0)
+          return 6 + lOffset;
+        if (cmsswdet == 4 && cmsswdisk == 3 && isStereo == 1)
+          return 7 + lOffset;
+        //TEC
+        if (cmsswdet == 6 && cmsswdisk == 1 && isStereo == 0)
+          return 8 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 1 && isStereo == 1)
+          return 9 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 2 && isStereo == 0)
+          return 10 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 2 && isStereo == 1)
+          return 11 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 3 && isStereo == 0)
+          return 12 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 3 && isStereo == 1)
+          return 13 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 4 && isStereo == 0)
+          return 14 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 4 && isStereo == 1)
+          return 15 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 5 && isStereo == 0)
+          return 16 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 5 && isStereo == 1)
+          return 17 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 6 && isStereo == 0)
+          return 18 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 6 && isStereo == 1)
+          return 19 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 7 && isStereo == 0)
+          return 20 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 7 && isStereo == 1)
+          return 21 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 8 && isStereo == 0)
+          return 22 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 8 && isStereo == 1)
+          return 23 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 9 && isStereo == 0)
+          return 24 + lOffset;
+        if (cmsswdet == 6 && cmsswdisk == 9 && isStereo == 1)
+          return 25 + lOffset;
+        return -1;
+      }
+    }
+    TkLayout lo_;
```

Comment:
private?

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit: 
 https://github.com/trackreco/cmssw/pull/76/commits/1abf1f1f3c212a125d5ae5470a91c1c1424a9e80